conjureup/controllers/bootstrap/common.py

```diff
@@ -0,0 +1,86 @@
+from pathlib import Path
+
+from conjureup import controllers, juju, utils
+from conjureup.app_config import app
+from conjureup.telemetry import track_event
+
+
+class BaseBootstrapController:
+    msg_cb = NotImplementedError()
+
+    def render(self):
+        app.loop.create_task(self.do_bootstrap(app.current_credential))
+        controllers.use('deploy').render()
+
+    async def do_bootstrap(self, creds):
+        if app.current_controller is None:
+            app.current_controller = "conjure-up-{}-{}".format(
+                app.current_cloud,
+                utils.gen_hash())
+
+        if app.current_model is None:
+            app.current_model = utils.gen_model()
+
+        if not app.is_jaas:
+            await self.pre_bootstrap()
+            app.log.info('Bootstrapping Juju controller.')
+            self.msg_cb('Bootstrapping Juju controller.')
```

Comment:
Not a blocker just a place holder, I think these should be replaced with self.emit